src/Workspaces/Core/Portable/Workspace/Workspace_Registration.cs

```diff
@@ -56,7 +56,12 @@ protected void UnregisterText(SourceTextContainer textContainer)
                 throw new ArgumentNullException(nameof(textContainer));
             }
 
-            GetWorkspaceRegistration(textContainer).SetWorkspaceAndRaiseEvents(null);
+            var registration = GetWorkspaceRegistration(textContainer);
+
+            if (registration.Workspace == this)
```

Comment:
Should this throw if you try to unregister a textContainer that is registered to a different workspace?